### PR TITLE
Add user picture, and treat a bot account appropriately

### PR DESCRIPTION
**Pull request by youtsumi**
Hi @kaakaa-san,

I saw your development progressing. This is nice. The original version of my change did not treat user's icon and a bot account appropriately. This PR addresses those issues.

I still have a difficulty to handle time stamp in the footer. I am wondering if Mattermost is capable to time zone in the footer.

### CODE REVIEW

**Comment by youtsumi** [comment]
Seems like MessageWillBePosted() is not called when posting in iPhone app. And occasionally it isn't called by Desktop app as well. 
Also I notice that there is a case that embedding a bot icon doesn't work. It seems to occur with incoming webhook accounts. Not sure why yet.

File: server/plugin/hooks.go [hunk 1]
```diff
@@ -56,19 +56,32 @@ func (p *SharePostPlugin) MessageWillBePosted(c *plugin.Context, post *model.Pos
...
+			botUser := model.BotFromUser(postUser)
+			AuthorName = botUser.DisplayName
+			AuthorIcon = fmt.Sprintf(fmtstmnt, *siteURL, botUser.UserId)
+		}
+
 		attachment := []*model.SlackAttachment{
 			{
-				Timestamp:  oldPost.CreateAt,
-				AuthorName: postUser.GetDisplayNameWithPrefix(model.SHOW_NICKNAME_FULLNAME, "@"),
+				Timestamp:  ateAt,
```
Comment (kaakaa) [inline server/plugin/hooks.go]: This line may cause the following error.

```
2020-05-03T17:08:21.002+0900    error   mlog/log.go:175 RPC call MessageWillBePosted to plugin failed.  {"plugin_id": "business.silly.quote", "error": "reading body EOF"}
2020-05-03T17:08:21.002+0900    info    go-plugin/stream.go:15  2020/05/03 17:08:21 rpc: gob error encoding body: gob: type not registered for interface: time.Time     {"plugin_id": "business.silly.quote", "source": "plugin_stderr"}
2020-05-03T17:08:21.061+0900    debug   app/post_metadata.go:113        Failed to get embedded content for a post       {"post_id": "xp5x6ksom3bizg37dcg35iz1gw", "error": "Get \"http://192.168.11.11:8065/tttt/pl/sy4k3pgpbbdczpkmpg61p71r3y\": address forbidden, you may need to set AllowedUntrustedInternalConnections to allow an integration access to your internal network"}
```

`Timestamp` can accept value of any type, but it must be `string` or `int64`.
https://github.com/mattermost/mattermost-server/blob/master/model/slack_attachment.go#L29

**Review by kaakaa — CHANGES_REQUESTED** [review]
Thanks @youtsumi!
Your code is so helpful. Also, I didn't catch things about posts from `bot`. Thank you very much.

However, when I tried it, I got an error and it didn't seem to work. I commented on the cause, can you have it fixed?

And in my iOS app, `MessageWillBePosted` works fine and expanded local link... 🤔  What version do you use? (I use iOS app v1.30.1. and Mattermost server run from [master branch](https://github.com/mattermost/mattermost-server))

**Comment by youtsumi** [comment]
Sorry about that. Fixed it.

As for the iPhone Mobile app issue, I am also using v1.30.1. And I encountered with the successful case. The behavior seems not stable...
Multiple posts also seem to fail to expand a link but not always.

**Comment by kaakaa** [comment]
Thanks! It works fine now 👍 

I'm sorry, but could you fix the formatting of the code? `gofmt` should be useful for you. `gofmt` is language-default formatter of golang, and should be installed by default. If `gofmt` is installed, `gofmt -w server/` overwrite existing code by formatted code.

And if you run `make check-style` in the root of this repository, you can run the linter including the format check (it should be run as a CI, but I didn't configure it yet).

---

> As for the iPhone Mobile app issue, I am also using v1.30.1. And I encountered with the successful case. The behavior seems not stable...
> Multiple posts also seem to fail to expand a link but not always.

Thanks for additional comments. It's weird...
I create new issue #2 for this, so let's continue on #2.

**Comment by youtsumi** [comment]
Wow, what a nice feature! Thanks for letting me know.

**Review by kaakaa — APPROVED** [review]
Great! Thank you for fixing 👍

**Pull request merged**